eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/NodeRelay.scala

```diff
@@ -0,0 +1,347 @@
+    failures match {
+      case Nil => None
+      case LocalFailure(_, BalanceTooLow) :: Nil if routingFeeHigh =>
+        // We have direct channels to the target node, but not enough outgoing liquidity to use those channels.
+        // The routing fee proposed by the sender was high enough to find alternative, indirect routes, but didn't yield
+        // any result so we tell them that we don't have enough outgoing liquidity at the moment.
+        Some(TemporaryNodeFailure)
+      case LocalFailure(_, BalanceTooLow) :: Nil => Some(TrampolineFeeInsufficient) // a higher fee/cltv may find alternative, indirect routes
+      case _ if routeNotFound => Some(TrampolineFeeInsufficient) // if we couldn't find routes, it's likely that the fee/cltv was insufficient
+      case _ =>
+        // Otherwise, we try to find a downstream error that we could decrypt.
+        val outgoingNodeFailure = failures.collectFirst { case RemoteFailure(_, e) if e.originNode == nextPayload.outgoingNodeId => e.failureMessage }
+        val otherNodeFailure = failures.collectFirst { case RemoteFailure(_, e) => e.failureMessage }
+        val failure = outgoingNodeFailure.getOrElse(otherNodeFailure.getOrElse(TemporaryNodeFailure))
+        Some(failure)
+    }
+  }
+
+}
+
+/**
+ * see https://doc.akka.io/docs/akka/current/typed/style-guide.html#passing-around-too-many-parameters
+ */
+class NodeRelay private(
+                         nodeParams: NodeParams,
+                         router: ActorRef,
+                         register: ActorRef,
+                         relayId: UUID,
+                         paymentHash: ByteVector32,
+                         context: ActorContext[NodeRelay.Command],
+                         adapters: typed.ActorRef[Any],
+                         fsmFactory: FsmFactory) {
+
+  import NodeRelay._
+
+
+  def apply(): Behavior[Command] =
+    Behaviors.receiveMessagePartial {
+      // We make sure we receive all payment parts before forwarding to the next trampoline node.
+      case WrappedNodeRelayPacket(IncomingPacket.NodeRelayPacket(add, outer, inner, next)) => outer.paymentSecret match {
+        case None =>
+          // TODO: @pm: maybe those checks should be done later in the flow (by the mpp FSM?)
+          context.log.warn("rejecting htlcId={}: missing payment secret", add.id)
+          rejectHtlc(add.id, add.channelId, add.amountMsat)
+          Behaviors.stopped
+        case Some(secret) =>
+          import akka.actor.typed.scaladsl.adapter._
+          context.log.debug("creating a payment FSM")
+          val mppFsm = context.actorOf(MultiPartPaymentFSM.props(nodeParams, add.paymentHash, outer.totalAmount, adapters.toClassic))
+          context.log.debug("forwarding incoming htlc to the payment FSM")
+          mppFsm ! MultiPartPaymentFSM.HtlcPart(outer.totalAmount, add)
+          receiving(Queue(add), secret, inner, next, mppFsm)
+        case _ => Behaviors.unhandled
```

Comment:
This case is unnecessary, everything is already handled

Reply:
Fixed with 5467945e8.